aiida/storage/psql_dos/migrator.py

```diff
@@ -70,6 +70,33 @@ class PsqlDosMigrator:
 
     def __init__(self, profile: Profile) -> None:
         self.profile = profile
+        self._engine = create_sqlalchemy_engine(self.profile.storage_config)
+        self._connection = None
+
+    def close(self) -> None:
```

Comment:
If this method is mainly intended to be used upon destruction of the migrator, you might consider making it a private method.

Reply:
But it needs to be called from the outside. In fact, it is the `PsqlDosBackend` that calls it when it itself has received the `close` call. The `PsqlDosBackend` is also the only class the should be calling the `PsqlDosMigrator`.